main.js

```diff
@@ -3,6 +3,54 @@
 
 'use strict';
 
+// The download dialog generates many style recalculations while attached to
+// the DOM, regardless of its display style, so instead of setting its display
+// style, we add/remove it to/from the DOM.
+var display = document.getElementById("display");
+
+var downloadDialog = document.getElementById('download-screen');
+display.removeChild(downloadDialog);
+downloadDialog.style.display = 'block';
+function showDownloadScreen() {
+  display.appendChild(downloadDialog);
+}
+function hideDownloadScreen() {
+  if (downloadDialog.parentNode) {
+    downloadDialog.parentNode.removeChild(downloadDialog);
+  }
+}
+
+var splashScreen = document.getElementById('splash-screen');
+display.removeChild(splashScreen);
+splashScreen.style.display = 'block';
+function showSplashScreen() {
+  display.appendChild(splashScreen);
+}
+function hideSplashScreen() {
+  if (splashScreen.parentNode) {
+    splashScreen.parentNode.removeChild(splashScreen);
+  }
+}
+
+function showBackgroundScreen() {
+  document.getElementById("background-screen").style.display = "block";
+}
+function hideBackgroundScreen() {
+  document.getElementById("background-screen").style.display = "none";
+}
+
+// The exit screen is hidden by default, and we only ever show it,
+// so we don't need a hideExitScreen function.
+function showExitScreen() {
+  document.getElementById("exit-screen").style.display = "block";
+}
+
+// If the document is hidden, then we've been started by an alarm and are in
```

Comment:
I can, but it calls *showBackgroundScreen*, which isn't defined until main.js. Shall I move the whole block of show/hide functions as well?